Replace sed hack with 2to3 everywhere

vibss2397 opened the pull request:
Replace sed hack with 2to3 everywhere, fixes #660 

Replaced sed imports with the usage of 2to3 tool everywhere, the changes were mostly in the dockerfiles but there were 1-2 shell scripts too.

bcipriano commented on ci/run_python_tests.sh:
```diff
@@ -10,7 +10,8 @@ python -m grpc_tools.protoc -I=proto/ --python_out=rqd/rqd/compiled_proto --grpc
 # Fix imports to work in both Python 2 and 3. See
 # <https://github.com/protocolbuffers/protobuf/issues/1491> for more info.
-sed -i 's/^\(import.*_pb2\)/from . \1/' pycue/opencue/compiled_proto/*.py rqd/rqd/compiled_proto/*.py
+2to3 -wn -f import pycue/opencue/compiled_proto//*_pb2*.py
```
Extra slash on this line.

vibss2397 replied:
i will fix this

bcipriano commented on rqd/Dockerfile:
```diff
@@ -38,7 +38,7 @@ RUN python3.6 -m grpc_tools.protoc \
 # Fix imports to work in both Python 2 and 3. See
 # <https://github.com/protocolbuffers/protobuf/issues/1491> for more info.
-RUN sed -i 's/^\(import.*_pb2\)/from . \1/' rqd/rqd/compiled_proto/*.py
+2to3 -wn -f import rqd/rqd/compiled_proto/*_pb2*.py
```
Just to check, did you test these changes by doing a local build of the Docker container?

For example for RQD: https://www.opencue.io/docs/getting-started/deploying-rqd/#option-2-building-and-running-rqd-from-source

I believe we may need to add `2to3` to [our top-level `requirements.txt`](https://github.com/AcademySoftwareFoundation/OpenCue/blob/master/requirements.txt) to ensure that library is available within our various environments.

vibss2397 replied:
no i did not, i'll build it and check and get back to you

bcipriano commented:
@vibss2397 Thanks!

Also, make sure to resolve the CLA check. The comment from the EasyCLA bot will walk you through that process.

bcipriano commented:
@vibss2397 Ping, do you plan to keep working on this?

vibss2397 commented:
Hi @bcipriano, sorry i have been busy the past week or 2. I am having finals this week and would love to work on it from next week, however if someone else is willing to complete it faster, you can let them do it, no problem.

bcipriano commented:
Ok no rush! Just making sure things haven't gone stale.

vibss2397 commented:
hey @bcipriano do i need to do a docker build on every folder like rqd, ci, pycue etc or does it happen globally for the entire project?

bcipriano commented:
Each component (pycue, rqd, cuegui, etc.) gets built separately.

Here's an example for RQD: https://www.opencue.io/docs/getting-started/deploying-rqd/#option-2-building-and-running-rqd-from-source. The `docker build` command there can be adapted for each component.

Probably good enough to just try 2-3 of them and make sure that's working -- CI will check the rest once this PR gets merged. (The CI pipeline that does the Docker builds only runs on commits to `master` at the moment unfortunately.)

vibss2397 commented:
@bcipriano so i ran the docker build command and it does not work without having 2to3 in the requirements.txt file, do we want to add that now? or have a pip install command in each dockerfile separately.

bcipriano commented:
Yes, please go ahead and add `2to3` to our top-level `requirements.txt` in this PR.

vibss2397 commented:
@bcipriano fixed all the problems

bcipriano commented:
Thanks!

Something strange has happened in this PR -- it's now showing lots of older commits which were already merged to master and the "Files changed" tab is showing similar, lots of older changes.

Curious, did you do something in your branch e.g. a rebase? I've seen this before but it's unclear to me why it happens, it would be a huge help if we could figure out why.

vibss2397 commented:
I forked this a while ago and it was lagging behind this master branch, when i pulled from that it seemed to cause all the issues, let me check how to fix this in git and if that does not happen do you mind if i create a new pull request maybe by forking again?

bcipriano commented:
Do you remember how you updated your fork / branch? Like the commands/process you used?

vibss2397 commented:
so i changed my remote from this to my own fork and i pulled from that which created some merge issues which i fixed and then pushed it on my own fork after which i think all this happened. I don't think these issues were before yesterday so i guess this caused it.

bcipriano commented:
Hm, ok yeah. I think you may want to create a new branch/PR that's fresh from master and re-send it. Sorry about that.

Here's some info about how I manage my fork in case it's helpful. We should probably expand our developer docs with more on this.

First, I have two remotes configured -- the main repo and my personal fork:
```
$ git remote -v
origin	git@github.com:bcipriano/OpenCue.git (fetch)
origin	git@github.com:bcipriano/OpenCue.git (push)
upstream	git@github.com:AcademySoftwareFoundation/OpenCue.git (fetch)
upstream	git@github.com:AcademySoftwareFoundation/OpenCue.git (push)
```

To update my fork, I merge `upstream` locally then push that to my fork:
```
# switch to master locally
git checkout master
# make sure my local master has everything from my fork
git pull
# fetch info from upstream
git fetch upstream
# merge upstream master into my local master
git merge upstream/master
# push my local master up to my fork
git push
```

Then in my local branches, I can merge master to update them:
```
# switch to my branch
git checkout some-work-branch
# pull in latest from master (local/fork/upstream should all match at this point)
git merge master
# push the branch to my fork
git push
```
